ykrt/src/mt.rs

```diff
@@ -347,12 +355,76 @@ impl MTThread {
                     }
                 }
             }
-            PHASE_TRACING => {
+            PHASE_TRACING | PHASE_TRACING_LOCK => {
+                if lp & PHASE_TAG == PHASE_TRACING_LOCK {
+                    // PHASE_TRACING_LOCK requires special handling, because another thread might
+                    // have grabbed it on this Location even though this thread is tracing this
+                    // location.
+
+                    if self.inner.tracer.is_none() {
+                        // If this thread isn't tracing anything, there's little point in spinning
+                        // as the thread tracing this location is still plausibly alive.
+                        return None;
+                    }
+
+                    // We wait to grab the lock.
+                    // FIXME: this is a terrible spinlock.
+                    while lp & PHASE_TAG == PHASE_TRACING_LOCK {
+                        lp = loc.state.load(Ordering::Acquire);
+                        yield_now();
+                    }
+
+                    if lp & PHASE_TAG != PHASE_TRACING {
+                        // While waiting for the Location to move beyond PHASE_TRACING_LOCK, we
+                        // probably raced with the thread tracing this location when.
```

Comment:
when... what? Incomplete comment?

Reply:
Fixed in https://github.com/softdevteam/yk/pull/199/commits/77e2bd236333ecb7a9fe79d44cb854f35ab6f067.